src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,319 @@
+    |
+    |The output file is a tab-delimited table with one record per aggregated cluster of pileups. Aggregated
+    |pileups are reported with the minimum and maximum (inclusive) coordinates of all pileups in the cluster, a
+    |possible putative structural variant event type supported by the pileups, and the sum of read support from all
+    |pileups in the cluster.
+    |""")
+class AggregateSvPileup
+(@arg(flag='i', doc="The input text file of pileups generated by SvPileup") input: FilePath,
+ @arg(flag='o', doc="The output file") output: FilePath,
+ @arg(flag='d', doc="Distance threshold below which to cluster breakpoints") maxDist: Int = 100,
+) extends SvTool with AggregateSvPileupDef {
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+    // Read pileups from input file
+    val pileups: Seq[BreakpointPileup] = Metric.read[BreakpointPileup](input)
+
+    // Group pileups by left and right contig and strand
+    val pileupsByContigStrand: Map[BreakpointContigOrientation, PileupGroup] =
+      pileups.groupMap(BreakpointContigOrientation(_))(identity)
+
+    // Within each contig+strand group, identify all immediate "neighbors" of each pileup (other pileups within the
+    // distance threshold)
+    val pileupToNeighbors: Map[BreakpointContigOrientation, Map[BreakpointPileup, PileupGroup]] =
+      pileupsByContigStrand.view.mapValues(AggregateSvPileup.pileupToNeighbors(_, maxDist)).toMap
+
+    // Within each contig+strand group, recursively build (transitive) clusters of neighboring pileups
+    val pileupClusters: Map[BreakpointContigOrientation, Seq[PileupGroup]] =
+      pileupToNeighbors.view.mapValues(AggregateSvPileup.toClusters).toMap
+
+    // Aggregate each cluster
+    val aggregatedPileups: Map[BreakpointContigOrientation, Seq[AggregatedBreakpointPileup]] =
+      pileupClusters.view.mapValues(_.map(AggregatedBreakpointPileup.apply)).toMap
+
+    // Write output file
+    Metric.write[AggregatedBreakpointPileup](output, aggregatedPileups.values.flatten)
+  }
+
+}
+
+
+object AggregateSvPileup extends AggregateSvPileupDef {
+
+  /**
+   * Maps each pileup to the set of other pileups within the distance threshold
+   * @param pileups Set of pileups to evaluate for pairwise neighbor relationships
+   * @param maxDist Maximum distance (inclusive) for the left and right coordinates to be considered a neighbor
+   * @return Map of pileup to the set of neighbors
+   */
+  private def pileupToNeighbors(pileups: PileupGroup, maxDist: Int): Map[BreakpointPileup, PileupGroup] = {
+    val map: mutable.Map[BreakpointPileup, PileupGroup] = mutable.Map.from(pileups.map((_, Seq())))
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.hashCode() < pileup2.hashCode() && isCluster(pileup1, pileup2, maxDist)) {
+        map.put(pileup1, map(pileup1).appended(pileup2))
+        map.put(pileup2, map(pileup2).appended(pileup1))
```

Comment:
Might consider putting `Seq.newBuilder` into the maps instead of `Seq` so you can just add records as you go.  Or if you don't care about ordering you could use `Lists` instead (Lists are Seqs) in which case you could:

1. Prepopulate your map with `pileups.map((_, Nil))`
2. Add things here by doing `map.put(pileup1, pileup2 :: map(pileup1))` to avoid all the extra allocation